From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 12 Mar 2021 17:31:56 +0100
Subject: [PATCH] Change LLVM targets

 - Change x86_64-unknown-linux-gnu to use x86_64-frugalware-linux
 - Change i686-unknown-linux-gnu to use i686-frugalware-linux

Add aliases to match.
---
 compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs   | 2 +-
 compiler/rustc_target/src/spec/mod.rs                      | 3 +++
 compiler/rustc_target/src/spec/x86_64_unknown_linux_gnu.rs | 2 +-
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs b/compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs
index 083c115d084..2c20bca80bc 100644
--- a/compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs
+++ b/compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs
@@ -8,7 +8,7 @@ pub fn target() -> Target {
     base.stack_probes = true;
 
     Target {
-        llvm_target: "i686-unknown-linux-gnu".to_string(),
+        llvm_target: "i686-frugalware-linux".to_string(),
         pointer_width: 32,
         data_layout: "e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-\
             f64:32:64-f80:32-n8:16:32-S128"
diff --git a/compiler/rustc_target/src/spec/mod.rs b/compiler/rustc_target/src/spec/mod.rs
index 8d72df6850f..367ceab88ee 100644
--- a/compiler/rustc_target/src/spec/mod.rs
+++ b/compiler/rustc_target/src/spec/mod.rs
@@ -1812,6 +1812,9 @@ macro_rules! target_aliases {
         }
 
         target_aliases! {
+            ("x86_64-frugalware-linux", "x86_64-unknown-linux-gnu"),
+            ("i686-frugalware-linux", "i686-unknown-linux-gnu"),
+
             // `x86_64-pc-solaris` is an alias for `x86_64_sun_solaris` for backwards compatibility reasons.
             // (See <https://github.com/rust-lang/rust/issues/40531>.)
             ("x86_64-pc-solaris", "x86_64-sun-solaris"),
diff --git a/compiler/rustc_target/src/spec/x86_64_unknown_linux_gnu.rs b/compiler/rustc_target/src/spec/x86_64_unknown_linux_gnu.rs
index f127dd49bc4..a958e4fc11d 100644
--- a/compiler/rustc_target/src/spec/x86_64_unknown_linux_gnu.rs
+++ b/compiler/rustc_target/src/spec/x86_64_unknown_linux_gnu.rs
@@ -8,7 +8,7 @@ pub fn target() -> Target {
     base.stack_probes = true;
 
     Target {
-        llvm_target: "x86_64-unknown-linux-gnu".to_string(),
+        llvm_target: "x86_64-frugalware-linux".to_string(),
         pointer_width: 64,
         data_layout: "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
             .to_string(),
